app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqHeaderCompliancePartiallyCompliantValidConnectionMethod.scala.txt

```diff
@@ -0,0 +1,63 @@
+@import uk.gov.hmrc.hmrcemailrenderer.templates.tdq.extraparams.ValidConnectionMethodBase64EncodedDetails
+
+@(params: Map[String, Any])Fraud prevention headers status report
+
+Dear @params("developerName")
+
+@defining(ValidConnectionMethodBase64EncodedDetails.parse(params("extraDetails").toString)) { extraDetails =>
+Our API call logs show that between @params("fromDate") and @params("toDate") fraud prevention headers were submitted from application '@params("applicationName")' (id: @params("applicationId")) to our VAT Making Tax Digital (MTD) API(s).
+
+Our checks show that the fraud prevention headers submitted between these dates do not meet the requirements of the fraud prevention header specification on the HMRC Developer Hub.
+
+The report below is generated based on our basic checks on the fraud prevention header values sent by your application, for example:
+
+is the value for Gov-Client-Public-IP an IP address?
+
+is the IP address value for Gov-Client-Public-IP a public IP?
+
+The report below shows results for requests that provided the value for Gov-Client-Connection-Method header as @extraDetails.connectionMethod. It does not include headers that appear to be valid.
+
+If your application sent additional requests with a different value for the Gov-Client-Connection-Method you will receive separate reports for each.
+
+Errors and/or warnings may relate to more than one version of your software as provided in the Gov-Vendor-Version header.
+
+Report
+
+'@params("applicationName")' submitted @extraDetails.requestCount requests between @params("fromDate") and @params("toDate").
+
+These are the headers where errors and/or warnings have been found:
+
+@for(headersValidation <- extraDetails.headerValidations) {
+
+@headersValidation.headerOrHeaders
```

Comment:
Probably more of a copy question, but do we want to have comma separated header names here or should we turn it into something more readable?

Reply:
Currently it's comma separated, I'm personally happy with it